src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -1,122 +1,139 @@
         {
-            if (section.Key is string stringValue8)
+            if (obj is null)
             {
-                key = stringValue8;
-                if (obj.TryGetValue(key, out Program.MyClass2? element) && element is not null)
-                {
-                    BindCore(section, ref element);
-                    obj[key] = element;
-                }
-                else
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            String key;
+            foreach (IConfigurationSection section in configuration.GetChildren())
+            {
+                if (section.Key is string stringValue3)
                 {
-                    element = new Program.MyClass2();
-                    BindCore(section, ref element);
-                    obj[key] = element;
+                    key = stringValue3;
+                    if (obj.TryGetValue(key, out Program.MyClass2? element) && element is not null)
+                    {
+                        BindCore(section, ref element);
+                        obj[key] = element;
+                    }
+                    else
+                    {
+                        element = new Program.MyClass2();
+                        BindCore(section, ref element);
+                        obj[key] = element;
+                    }
                 }
             }
         }
-    }
 
-    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass2 obj)
-    {
-        if (obj is null)
+        public static void BindCore(IConfiguration configuration, ref Program.MyClass obj)
         {
-            throw new global::System.ArgumentNullException(nameof(obj));
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            if (configuration["MyString"] is string stringValue6)
+            {
+                obj.MyString = stringValue6;
+            }
+
+            if (configuration["MyInt"] is string stringValue7)
+            {
+                obj.MyInt = int.Parse(stringValue7);
+            }
+
+            List<Int32> temp8 = obj.MyList;
+            temp8 ??= new List<Int32>();
+            BindCore(configuration.GetSection("MyList"), ref temp8);
+            obj.MyList = temp8;
+
+            Dictionary<String, String> temp9 = obj.MyDictionary;
+            temp9 ??= new Dictionary<String, String>();
+            BindCore(configuration.GetSection("MyDictionary"), ref temp9);
+            obj.MyDictionary = temp9;
+
+            Dictionary<String, Program.MyClass2> temp10 = obj.MyComplexDictionary;
+            temp10 ??= new Dictionary<String, Program.MyClass2>();
+            BindCore(configuration.GetSection("MyComplexDictionary"), ref temp10);
+            obj.MyComplexDictionary = temp10;
         }
 
-    }
+        public static bool HasValue(IConfiguration configuration)
```

Comment:
This method doesn't appear to have callers. Can we only generate it when it has callers?